### PR TITLE
Kn 180

### PR DESCRIPTION
**Pull request by FredPeal**

### CODE REVIEW

File: src/mutations/user-interaction.mutation.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { gql } from '@apollo/client/core';
+
+export const USER_LIKE_ENTITY = gql`
+    mutation($input: UserInteractionInput!) {
+        userLikeEntity(input: $input)
+    }
+`;
+
+export const USER_UN_lIKE_ENTITY = gql`
```
Comment (RivierGrullon) [inline src/mutations/user-interaction.mutation.ts:9]: Suggestion: we can named these mutations without separate the word like `USER_UNLIKE_ENTITY` and at the end of the mutations we add "_MUTATION"
`USER_UNLIKE_ENTITY_MUTATION`

File: src/modules/user-interaction/index.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    ResponseUserDislikeEntity,
+    UserInteraction,
+    EntityInteraction
+} from '../../types/';
+import { ClientType } from '../../index';
+import { GET_USERS_INTERACTIONS_QUERY, GET_USER_INTERACTION_QUERY } from '../../queries';
+export class UsersInteractions {
+
+    constructor(protected client: ClientType) {}
+
```
Comment (RivierGrullon) [inline src/modules/user-interaction/index.ts:19]: Suggestion: Format this file pls

File: src/types/user-interaction.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    user?: UserData;
+    interactions: Interactions;
+    entity_id: string;
+    entity_namespace: string;
+    interactions_id: string;
+}
+export interface EntityInteraction {
+    entity_id: string;
+    entity_namespace: string;
+    interactions: string;
```
Comment (RivierGrullon) [inline src/types/user-interaction.ts:20]: Question: Is this interaction field not a type of the interface Interactions?

**Review by RivierGrullon — CHANGES_REQUESTED** [review]

File: src/mutations/user-interaction.mutation.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    mutation($input: UserInteractionInput!) {
+        userUnLikeEntity(input: $input)
+    }
+`;
+
+export const  USER_DIS_LIKE_ENTITY = gql`
```
Comment (RivierGrullon) [inline src/mutations/user-interaction.mutation.ts:15]: Question: what's the difference in user interaction between UnLike and DisLike?

**Review by RivierGrullon — CHANGES_REQUESTED** [review]

File: src/modules/user-interaction/index.ts (hunk 2 above)
Comment (FredPeal) [reply src/modules/user-interaction/index.ts:19]: I don't have formmater, can you send me one

File: src/mutations/user-interaction.mutation.ts (hunk 4 above)
Comment (FredPeal) [reply src/mutations/user-interaction.mutation.ts:15]: unLike is remove previous like , Dislike is another interaction, is as "I don't like this post"

**Review by RivierGrullon — APPROVED** [review]

**Pull request merged**